timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -484,6 +487,18 @@ public void stressTestForPaxosEndpoints() {
         }
     }
 
+    @Test
+    public void leaderRespondsToMultiClientRequests() {
+        MultiClientConjureTimelockService multiClientConjureTimelockService =
+                cluster.currentLeaderFor(client.namespace()).multiClientService();
+        Set<String> expectedNamespaces = ImmutableSet.of("alpha", "beta");
+        List<NamespacedLeaderTime> leaderTimes =
+                multiClientConjureTimelockService.leaderTimes(AuthHeader.valueOf("Bearer omitted"), expectedNamespaces);
+        Set<String> namespaces =
+                leaderTimes.stream().map(NamespacedLeaderTime::getNamespace).collect(Collectors.toSet());
+        assertThat(namespaces).hasSameElementsAs(expectedNamespaces);
```

Comment:
I don't know leader time (and/or the internals of Timelock) all that well - would leader times from different namespaces have different leadership IDs, or the same? If you expect one or the other, can you verify that in this test?

Reply:
Leadership ids should be different as each client will have its own instance of AsyncTimelockService -> AsyncLockService -> LeaderClock. I've added the check now.